### ISSUE TITLE
[BUG] It doesn't work 

### ISSUE THREAD

**Issue by moshpirit** [0x7b1/logseq-plugin-automatic-url-title#19]
I've disabled all plugins, restarted it and refreshed it after realizing that it doesn't work (I'm trying to parse `https://gitlab.com`, and `https://youtube.com`)  I'm using 0.10.5 (flatpak). It loads the icon, but not the title of the URL.

**Comment by hmamigo** [comment]
I have the same problem. I use Logseq 0.10.4 (appimage) on Linux.

**Comment by farfoodyou** [comment]
me too. who can fix this bug?

**Comment by CR0CKER** [comment]
Yep, it's broken. Too bad -- none of the other plugins available seem to do this well either.

**Comment by PwrSrg** [comment]
Confirmed **BROKEN**.

**Comment by junheep-dev** [comment]
Hi guys,
I've also discovered that the plugin is broken. As this repository is no longer maintained, I forked it and made a small fix.
You can find my release [here](https://github.com/wns3645/logseq-plugin-automatic-url-title/releases)

**Comment by fanyangto** [comment]
> Hi guys, I've also discovered that the plugin is broken. As this repository is no longer maintained, I forked it and made a small fix. You can find my release [here](https://github.com/wns3645/logseq-plugin-automatic-url-title/releases)

Thank you thank you for fixing it!

One question. How do I trigger the URL title download without hitting a return ? In the plugin URL Hyperlink, as soon as  URL is pasted, the download starts. It can also be re-triggered by adding / or any trivial change to the URL.

**Comment by fanyangto** [comment]
> Hi guys, I've also discovered that the plugin is broken. As this repository is no longer maintained, I forked it and made a small fix. You can find my release [here](https://github.com/wns3645/logseq-plugin-automatic-url-title/releases)

After using it for a few days, I found this plugin has a design issue. It appears that the plugin retrieval of URL is triggered whenever the URL rolls into view (scrolling down a long page). This consumes huge amount of resources, especially in a page with dozens of tagged references (e.g. project A page is referenced in dozens of journal pages). It's so intense that you can actually see the icon is flashing (delay in the download). 

The older plugin I used (URL hyperlink) doesn't suffer from this problem. The download of the URL only occurs when a URL is entered or modified. Unfortunately I deleted the plugin after using it for a long time (months). The only complaint  I have is that the retrieved title often contains &quote, and other encoded characters. But it never had any stability issue or consuming a lot of power. 

I hope someone still has the URL hyperlink plugin to allow us to replicate its codes. The plugin while still in the marketplace, is no longer downloadable.

**Comment by junheep-dev** [comment]
@fanyangto 
Did the URL hyperlink plugin also have the feature to display a favicon? I can't check how it works because it has shut down.

Because of the implementation of this plugin, blinking favicon is unavoidable whenever the block is rendered. But it is not very harmful to the performance because it does not download the URL and favicon again if it has already been fetched before. 
- If the link is already formatted link (like `[]()` format), it does not download the title of the link again.
- If it downloads the favicon image once, it uses the favicon from the browser cache next time. So, download via network will not be invoked

**Comment by fanyangto** [comment]
> @fanyangto Did the URL hyperlink plugin also have the �feature to display a favicon? I can't check how it works because it has shut down.
> 
> Because of the implementation of this plugin, blinking favicon is unavoidable whenever the block is rendered. But it is not very harmful to the performance because it does not download the URL and favicon again if it has already been fetched before.
> 
>     * If the link is already formatted link (like `[]()` format), it does not download the title of the link again.
> 
>     * If it downloads the favicon image once, it uses the favicon from the browser cache next time. So, download via network will not be invoked

URL hyperlink does not have the icon display. I personally think the icon feature is not that important and should have a switch to turn it off in your plugin. 

Yes, the URL hyperlink's github account was accidentally turned off. The author is trying to restore it. I requested and he sent me the binaries on a logseq forum. Now I have restored URL Hyperlink restored, and everything is back to normal. 

Here is the link he shared for the binary download: 

https://discuss.logseq.com/t/plugin-url-hyperlink-no-longer-available/28702/2

I personally think instead of everyone creates their own version of this simple but useful plugin. It would be far better to take the best and contribute and make it better. In my view, after trying all the URL title retrieval plugins, URL Hyperlink is the best and the only bug (the author acknowledges) is that sometimes it retrieves back titles with special characters encoded. But that I think there should be an easy fix.  

If you think icons are useuful, by all means make the feature optional.

**Comment by kawpii** [comment]
There's new plugin on marketplace: https://github.com/imjn/logseq-url-title-tagger

**Comment by fanyangto** [comment]
> There's new plugin on marketplace: https://github.com/imjn/logseq-url-title-tagger

Thanks for the suggestion. The new plugin works well on the surface. But it has a bug makes it next to impossible to use. It repeatedly load links so on a page with a lot of links (e.g. a tag page where the daily journals are referenced over 100 times, the page loading and scrolling slows down to a halt. I opened an issue last year against the plugin. It still hasn't been fixed (just tried again).

So far the only plugin that works consistently is "URL hyperlink."

**Comment by kawpii** [comment]
> > There's new plugin on marketplace: https://github.com/imjn/logseq-url-title-tagger
> 
> Thanks for the suggestion. The new plugin works well on the surface. But it has a bug makes it next to impossible to use. It repeatedly load links so on a page with a lot of links (e.g. a tag page where the daily journals are referenced over 100 times, the page loading and scrolling slows down to a halt. I opened an issue last year against the plugin. It still hasn't been fixed (just tried again).
> 
> So far the only plugin that works consistently is "URL hyperlink."

I didn't thought it was fork of this one, I installed the plugin you suggested, thank you!
